src/rust/engine/src/externs.rs

```diff
@@ -529,34 +546,42 @@ pub struct HandleBuffer {
   handle_: Handle,
 }
 
-impl HandleBuffer {
-  pub fn to_vec(&self) -> Vec<Value> {
-    with_vec(self.handles_ptr, self.handles_len as usize, |handle_vec| {
-      handle_vec
-        .iter()
-        .map(|h| Value::new(unsafe { h.clone_shallow() }))
-        .collect()
-    })
+impl RawBuffer<Handle, Value> for HandleBuffer {
+  fn ptr(&self) -> *mut Handle {
+    self.handles_ptr
   }
 
-  ///
-  /// Asserts that the HandleBuffer contains one value, and returns it.
-  ///
-  /// NB: Consider making generic and merging with TypeIdBuffer if we get a third copy.
-  ///
-  pub fn unwrap_one(&self) -> Value {
-    assert!(
-      self.handles_len == 1,
-      "HandleBuffer contained more than one value: {}",
-      self.handles_len
-    );
-    with_vec(self.handles_ptr, self.handles_len as usize, |handle_vec| {
-      Value::new(unsafe { handle_vec.iter().next().unwrap().clone_shallow() })
-    })
+  fn len(&self) -> u64 {
+    self.handles_len
+  }
+
+  fn lift(t: &Handle) -> Value {
+    Value::new(unsafe { t.clone_shallow() })
+  }
+}
+
+#[repr(C)]
+pub struct IdentBuffer {
+  idents_ptr: *mut Ident,
+  idents_len: u64,
+  // A Handle to hold the underlying array alive.
+  handle_: Handle,
```

Comment:
I know this part isn't new, but could we add a comment on one of these fields explaining why `handle_` needs to exist? Is it so python doesn't gc the pointer?

Reply:
That's correct. Since CI is green here, I'm going to go ahead and merge this one, but can revisit.